engine/src/exec-interface-object.cpp

```diff
@@ -2462,7 +2462,7 @@ void MCObject::SetPattern(MCExecContext& ctxt, uint2 p_new_pixmap, uint4* p_new_
         else
             if (t_isopened)
                 MCpatternlist->freepat(patterns[i].pattern);
-        if (*p_new_id < PI_PATTERNS)
+        if (*p_new_id <= PI_END - PI_PATTERNS)
             *p_new_id += PI_PATTERNS;
```

Comment:
After investigating MetaCard I believe the documented range should be 137 to 300